### PR TITLE
fix: show modal dialog for those that need user action

### PR DESCRIPTION
**Pull request by HuihuiWu-Microsoft**
ADO: 9951504

image

### CODE REVIEW

File: packages/fx-core/src/core/index.ts [hunk 1]
```diff
@@ -271,7 +271,8 @@ class CoreImpl implements Core {
             new DialogMsg(DialogType.Show, {
               description: `Download '${samples.label}' from Github. This will download '${samples.label}' repository and open to your local machine`,
               level: MsgLevel.Info,
-              items: ["Download", "Cancel"],
+              items: ["Download"],
```
Comment (jiayi-ruan) [inline packages/fx-core/src/core/index.ts:274]: CLI need to be changed since the cancel button is removed. pls wait for my commit later, thx.

**Review by 1openwindow — APPROVED** [review]

**Comment by 1openwindow** [comment]
LGTM

**Review by nliu-ms — APPROVED** [review]

File: packages/fx-core/resource/strings.json [hunk 2]
```diff
@@ -2,14 +2,15 @@
   "core": {},
   "solution": {
     "ScaffoldSuccessNotice": "[Teams Toolkit] Scaffold successfully.",
-    "ProvisionConfirmNotice": "The Teams Toolkit will provision resources under %s (Azure account) with %s (subscription). Costs may incur according to the [cost calculator](https://azure.microsoft.com/en-us/pricing/calculator/).",
+    "ProvisionConfirmNotice": "The Teams Toolkit will provision resources under %s (Azure account) with %s (subscription). Costs may incur according to the cost calculator.",
```
Comment (1openwindow) [inline packages/fx-core/resource/strings.json:5]: should be "pricing calculator"

**Comment by Alive-Fish** [comment]
Test Tab with SQL and Function by CLI:
image

Deploy without provisioning:
image

Provision:
image

Provision twice:
image

Deploy:

Deploy twice:

Validate/Build:

Publish:

Publish twice:

**Review by chagong — APPROVED** [review]

**Review by MSFT-yiz — APPROVED** [review]

**Pull request merged**